pkg/models/model_gallery.go

```diff
@@ -111,6 +111,42 @@ func (g *Gallery) listZipContents() ([]*zip.File, *zip.ReadCloser, error) {
 		b := filteredFiles[j]
 		return utils.NaturalCompare(a.Name, b.Name)
 	})
+	cover := Contains(filteredFiles, "cover.jpg") // first image with cover.jpg in the name
+	if cover >= 0 {                               // will be moved to the start
+		reorderedFiles := Reorder(filteredFiles, cover)
+		if reorderedFiles != nil {
+			return reorderedFiles, readCloser, nil
+		}
+	}
 
 	return filteredFiles, readCloser, nil
 }
+
+func Contains(a []*zip.File, x string) int { // return index of first occurenece of string x in name of zip contents
+	for i, n := range a { // -1 otherwise
+		if strings.Contains(n.Name, x) {
+			logger.Debugf("Cover found in zip file %s.Position:%d", n.Name, i)
+			return i
+		}
+	}
+	return -1
+}
+
+func Reorder(a []*zip.File, toFirst int) []*zip.File { // reorder slice so that element with position toFirst gets at the start
```

Comment:
As above. I think there may be a more efficient way of doing this via slice ranges instead of iterating through the entire slice.

Reply:
The order needs to be kept after the removal of the element so a copy is needed. I doubt that the difference will be that much. Nevertheless  fixing that also.